ckanext/external_storage/blueprints.py

```diff
@@ -1,39 +1,64 @@
 """ckanext-external-storage Flask blueprints
 """
+import logging
+
+from ckan.common import request
 from ckan.plugins import toolkit
 from flask import Blueprint
 
 from .download_handler import call_download_handlers, call_pre_download_handlers, get_context
 
+log = logging.getLogger(__name__)
+
 blueprint = Blueprint(
     'extstorage',
     __name__,
 )
 
 
-def download(id, resource_id, filename=None):
+def download(dataset_id, resource_id, filename=None):
     """Download resource blueprint
 
     This calls all registered download handlers in order, until
     a response is returned to the user
     """
+    activity_id = request.args.get('activity_id')
```

Comment:
It's better to use `toolkit.request.params.get('activity_id')`
```suggestion
    activity_id = toolkit.request.params.get('activity_id')
```